File: mapmint-services/raster-tools-src/cgi-env/service.py
import zoo
import os, sys, shutil


def copyTileIndex(conf, inputs, outputs):
    dirs = os.listdir(inputs["InputDSTN"]["value"])
    lname = None
    for i in dirs:
        ii = i.split('.')
        zoo.info(str(ii) + " " + inputs["InputDSON"]["value"])
        if ii[0] == inputs["InputDSON"]["value"]:
            dirs = [inputs["InputDSTN"]["value"] + "/" + i]
            try:
                zoo.info(inputs["InputDSTN"]["value"] + "/" + i)
                zoo.info(inputs["InputDSTN"]["value"] + "/" + i)
                os.unlink(inputs["InputDSTN"]["value"] + "/" + i)
            except Exception as e:
                zoo.error(str(e))
                pass
            lname = ii[0]
            lfname = i
            break
    lname = inputs["InputDSON"]["value"]
    try:
        dirs = os.listdir(conf["main"]["tmpPath"] + "/TEMP_" + lname)
        for i in dirs:
            zoo.info(conf["main"]["tmpPath"] + "/TEMP_" + lname + "/" + i, conf["main"]["dataPath"] + "/dirs/" + inputs["idir"]["value"] + "/" + i)
            shutil.move(conf["main"]["tmpPath"] + "/TEMP_" + lname + "/" + i, conf["main"]["dataPath"] + "/dirs/" + inputs["idir"]["value"] + "/" + i)
            try:
                os.unlink(conf["main"]["dataPath"] + "/dirs/" + inputs["idir"]["value"] + "/ds_ows.map")
            except:
                pass
    except Exception as e:
        zoo.error(str(e))
        zoo.error(conf["main"]["tmpPath"] + "/TEMP_" + lfname + " " + dirs[0])
        shutil.move(conf["main"]["tmpPath"] + "/TEMP_" + lfname, dirs[0])
        try:
            os.unlink(dir[0].replace(lfname, "ds_ows.map"))
        except Exception as e:
            zoo.error(str(e))
            pass

        pass
    outputs["Result"]["value"] = zoo._("Tile Index moved")
    return 3

File: mapmint-services/raster-tools-src/cgi-env/service.js
function createTindex(conf,inputs,outputs){
  
  var myOutputs= {Result: { type: 'RawDataOutput', "mimeType": "application/json" }};
  var myProcess = new ZOO.Process(conf["main"]["serverAddress"],'raster-tools.tindex');
  inputs["iname"]["value"]="tile_"+inputs["iname"]["value"];
  inputs["dir"]["value"]=inputs["dir"]["value"].replace(conf["main"]["dataPath"]+"/ftp/","")
  var myExecuteResult=myProcess.Execute(inputs,myOutputs);

  alert(myExecuteResult);

  inputs["InputDSTN"]={"value": conf["main"]["dataPath"]+"/dirs/"+inputs["idir"]["value"]}
  inputs["InputDSON"]={"value": inputs["iname"]["value"]}
  inputs["a_srs"]=inputs["srs"];
  inputs["a_srs"]["value"]="+init="+inputs["a_srs"]["value"];
  inputs["OutputDSN"]={"value": "TEMP_"+inputs["iname"]["value"]}

  var myProcess1 = new ZOO.Process(conf["main"]["serverAddress"],'vector-converter.Converter');
  var myOutputs1= {Result: { type: 'RawDataOutput', "mimeType": "application/json" }};
  var myExecuteResult1=myProcess1.Execute(inputs,myOutputs1);

  
  alert(myExecuteResult1);

  var myProcess2 = new ZOO.Process(conf["main"]["serverAddress"],'raster-tools.copyTileIndex');
  var myOutputs2= {Result: { type: 'RawDataOutput', "mimeType": "application/json" }};
  var myExecuteResult2=myProcess2.Execute(inputs,myOutputs2);

  alert(myExecuteResult2);

    return {result: 3, outputs: {"Result": {"value": ZOO._("TileIndex file created")}}};
}
